### PR TITLE
Winit multiwindow manager

### PR DESCRIPTION
**Pull request by j4qfrost**
Merry Christmas! At least it still is in PST. This draft has some of my initial ideas for multiple windows.I had to throw things together a bit haphazardly, because I had been away from the project a while. It runs with ```--features=winit --no-default-features```, but is broken for sdl

### CODE REVIEW

File: src/bridge/layouts/winit.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
+use super::unsupported_key;
```
Comment (Kethku) [inline src/bridge/layouts/winit.rs:1]: Layouts got moved to the window folder which is where they should have been to start, I just had them in the bridge for historical reasons. So these files aren't necessary anymore

File: src/bridge/ui_commands.rs [hunk 2]
```diff
@@ -112,25 +112,25 @@ impl UiCommand {
                 if unregister_rightclick() {
                     let msg = "Could not unregister previous menu item. Possibly already registered or not running as Admin?";
                     nvim.err_writeln(msg).await.ok();
-                    error!("{}", msg);
```
Comment (Kethku) [inline src/bridge/ui_commands.rs:115]: I'm curious about this change. Wouldn't we want to log these as errors to the console if the registration fails?

File: src/renderer/mod.rs [hunk 3]
```diff
@@ -59,11 +60,11 @@ pub struct Renderer {
     pub font_width: f32,
     pub font_height: f32,
     pub window_regions: Vec<WindowDrawDetails>,
-    pub batched_draw_command_receiver: Receiver<Vec<DrawCommand>>,
+    pub batched_draw_command_receiver: Option<Arc<Receiver<Vec<DrawCommand>>>>,
 }
 
 impl Renderer {
-    pub fn new(batched_draw_command_receiver: Receiver<Vec<DrawCommand>>) -> Renderer {
```
Comment (Kethku) [inline src/renderer/mod.rs:66]: Minor tidbit, but I think the default version of new should take the draw command receiver as I see the window as requiring it to function normally. But clearly you need a version of the renderer which doesn't have it hooked up to start, so I think we should have a separate new, maybe "new_unhooked" which constructs the renderer without the draw command channel? What do you think?

File: src/settings/mod.rs [hunk 4]
```diff
@@ -12,6 +12,7 @@ use nvim_rs::Neovim;
 use parking_lot::RwLock;
 pub use rmpv::Value;
 use tokio::process::ChildStdin;
+#[cfg(windows)]
```
Comment (Kethku) [inline src/settings/mod.rs:15]: I like this change

File: src/window/keyboard.rs [hunk 5]
```diff
@@ -1,81 +1,81 @@
...
-        result = format!("D-{}", result);
-    }
-
-    if special {
-        result = format!("<{}>", result);
-    }
-
-    result
-}
+use log::error;
```
Comment (Kethku) [inline src/window/keyboard.rs:1]: I assume this is diffed due to newlines? I reformatted my machine mid PR and things got a little weird :l

File: src/window/mod.rs [hunk 6]
```diff
@@ -101,20 +101,13 @@ pub fn create_window(
 ) {
     let (width, height) = window_geometry_or_default();
 
-    let renderer = Renderer::new(batched_draw_command_receiver);
```
Comment (Kethku) [inline src/window/mod.rs:104]: I'm curious about the moving of the size here. Can you give some details?

**Comment by Kethku** [comment]
@j4qfrost this pretty out of date. Are you still planning on coming back to this or should I close it as stale?

**Comment by j4qfrost** [comment]
I don't mind closing it, I think even while I was working on it the overall
project had a different trajectory. I've been distracted by other things,
will be back eventually. Hope you guys are well.

**Pull request closed without merging**